engine/src/main/java/com/redhat/cloud/notifications/processors/webhooks/WebhookTypeProcessor.java

```diff
@@ -109,72 +118,70 @@ private WebClient getWebClient(boolean disableSSLVerification) {
-                                // must be manually re-enabled
-                                // Redirects etc should have been followed by the vertx (test this)
-                                LOGGER.debugf("Webhook request to %s failed: %d %s %s", reqImpl.host(), resp.statusCode(), resp.statusMessage(), json);
-                                history.setInvocationResult(false);
-                            }
-
-                            if (!history.isInvocationResult()) {
-                                JsonObject details = new JsonObject();
-                                details.put("url", getCallUrl(reqImpl));
-                                details.put("method", reqImpl.method().name());
-                                details.put("code", resp.statusCode());
-                                // This isn't async body reading, lets hope vertx handles it async underneath before calling this apply method
-                                details.put("response_body", resp.bodyAsString());
-                                history.setDetails(details.getMap());
-                            }
-
-                            if (serverError) {
-                                throw new ServerErrorException(history);
-                            }
-                            return history;
-                        })
-                        .onFailure(this::shouldRetry)
-                        .retry().withBackOff(initialRetryBackOff, maxRetryBackOff).atMost(maxRetryAttempts)
-                        .onFailure().recoverWithItem(t -> {
-
-                            if (t instanceof ServerErrorException) {
-                                return ((ServerErrorException) t).getNotificationHistory();
-                            }
-
-                            NotificationHistory history = buildNotificationHistory(item, startTime);
-
-                            HttpRequestImpl<Buffer> reqImpl = (HttpRequestImpl<Buffer>) req.getDelegate();
-
-                            LOGGER.debugf("Failed: %s", t.getMessage());
-
-                            // TODO Duplicate code with the error return code part
-                            JsonObject details = new JsonObject();
-                            details.put("url", reqImpl.uri());
-                            details.put("method", reqImpl.method());
-                            details.put("error_message", t.getMessage()); // TODO This message isn't always the most descriptive..
-                            history.setDetails(details.getMap());
-                            return history;
-                        })
-                );
+        try {
+            return Failsafe.with(retryPolicy).get(() -> {
+
+                // TODO NOTIF-488 We may want to move to a non-reactive HTTP client in the future.
+                HttpResponse<Buffer> resp = req.sendJsonObject(payload).await().indefinitely();
```

Comment:
Should we have a reasonable timeout here?

Reply:
Good catch! Yes we need a timeout here.

Reply:
I added a temporary timeout value (60 seconds) which can be changed with configuration. I'm not sure about the best value there... don't hesitate to suggest something else.

I'll try to remove that `await()` call soon so this should only be temporary hopefully.